### ISSUE TITLE
[2015.5.3][raet] raet error with SaltRaetRoadStackJoiner

### ISSUE THREAD

**Issue by gtmanfred** [saltstack/salt#25432]
@thatch45 Can yall take a look at this?

https://bugs.archlinux.org/task/45647

Looking at the _requirements.txt and the raet-requirements.txt, it looks like m2crypto has been removed from the dependencies for raet?

Thanks
Daniel

**Comment by gtmanfred** [comment]
https://github.com/saltstack/salt/blob/2015.5/_requirements.txt
https://github.com/saltstack/salt/blob/2015.5/raet-requirements.txt

**Comment by jfindlay** [comment]
@gtmanfred, m2crypto has been deprecated in salt as of Beryllium (2015.8).  The only remaining depender is the x509 execution module.  In fact RAET never needed m2crypto.

**Comment by jfindlay** [comment]
It would be helpful to know which `RSA` is triggering that error, because it very likely shouldn't happen.

**Comment by gtmanfred** [comment]
Looks like I didn't package transport.conf with the minion.d directory.

But with I am getting a new error now.

[ERROR   ] 'SaltRaetRoadStackJoiner' object has no attribute 'masters'

```
[root@arch ioflo]# salt-call --versions-report
           Salt: 2015.5.3-79-gb9abd72
         Python: 2.7.10 (default, May 26 2015, 04:16:29)
         Jinja2: 2.7.3
       M2Crypto: Not Installed
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: 1.4.3
         PyYAML: 3.11
          ioflo: 1.3.8
          PyZMQ: Not Installed
```
           RAET: 0.6.3
            ZMQ: Not Installed
           Mako: Not Installed
        Tornado: 4.2

**Comment by gtmanfred** [comment]
```
[root@arch ioflo]# salt-minion -l debug
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/transport.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/transport.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: arch.gtmanfred.com
[DEBUG   ] Configuration file path: /etc/salt/minion
[INFO    ] Setting up the Salt Minion "arch.gtmanfred.com"
[DEBUG   ] Created pidfile: /var/run/salt-minion.pid
[INFO    ] The salt minion is starting up
[DEBUG   ] RAETEvent Using Jobber Stack at = /var/run/salt/minion/arch.gtmanfred.com_minion.event51b19bec8342348aca.uxd

[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/transport.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/transport.conf
[DEBUG   ] False
[INFO    ] The salt minion is shut down
[ERROR   ] 'SaltRaetRoadStackJoiner' object has no attribute 'masters'
[WARNING ] ** Restarting minion **
[INFO    ] Sleeping random_reauth_delay of 8 seconds
```

**Comment by gtmanfred** [comment]
https://github.com/saltstack/salt/blob/2015.5/salt/daemons/flo/core.py#L256

On this line, self.masters isn't populated.

**Comment by gtmanfred** [comment]
I just verified, everything works in develop

**Comment by gtmanfred** [comment]
It looks like this is happening because the postinitio stuff is not getting run for some reason.

**Comment by gtmanfred** [comment]
This was broken and explained here.

https://github.com/saltstack/salt/issues/21640

Can yall backport a fix to 2015.5?

**Comment by gtmanfred** [comment]
nevermind, I am doing it.

**Comment by gtmanfred** [comment]
closed by #25525
